framework/src/modules/random/utils.ts

```diff
@@ -108,14 +108,16 @@ export const bitwiseXOR = (bufferArray: Buffer[]): Buffer => {
 		return bufferArray[0];
 	}
 
-	const bufferSizes = new Set(bufferArray.map(buffer => buffer.length));
-	if (bufferSizes.size > 1) {
-		throw new Error('All input for XOR should be same size');
+	const size = bufferArray[0].length;
```

Comment:
it looks like this function is always called with bufferArray either with length 2.
Changing the signature to 
`bitwiseXOR = (bufferArray: [Buffer, Buffer]): Buffer` will introduce more efficient path

Reply:
Good point 👍🏻 

I think that neither `getRandomSeed()` function nor the tests utilize `bitwiseXOR()` function correctly.

Please take a look at the new commit. I believe I have provided much cleaner and faster implementation.